src/Umbraco.Forms.Integrations.Commerce.EMerchantPay/Controllers/EmerchantPayController.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+#if NETCOREAPP
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+
+using Umbraco.Cms.Web.BackOffice.Controllers;
+using Umbraco.Cms.Web.Common.Attributes;
+#else
+using System.Web;
+using System.Web.Http;
+
+using Umbraco.Web.Mvc;
+using Umbraco.Web.WebApi;
+#endif
+using Newtonsoft.Json;
+
+using Umbraco.Forms.Integrations.Commerce.EMerchantPay.Models.Dtos;
+
+
+namespace Umbraco.Forms.Integrations.Commerce.EMerchantPay.Controllers
+{
+    [PluginController("UmbracoFormsIntegrationsCommerceEmerchantPay")]
+    public class EmerchantPayController : UmbracoAuthorizedApiController
+    {
+        private static readonly ReaderWriterLockSlim _lock = new ReaderWriterLockSlim();
+
+#if NETCOREAPP
+        private readonly IWebHostEnvironment _webHostEnvironment;
+
+        public EmerchantPayController(IWebHostEnvironment webHostEnvironment)
+        {
+            _webHostEnvironment = webHostEnvironment;
+        }
+#endif
+
+        [HttpGet]
+        public IEnumerable<CurrencyDto> GetCurrencies()
```

Comment:
It's good that these are coming from a file now rather than being hard-coded, but think we still have an issue if a user wants to have different currencies available.  With Umbraco 9 at least, they couldn't really edit this file, as a rebuild would remove and recreate it.

So maybe better in an actual config file?  For Umbraco 9 and .NET 5, we can have complex types fairly easily within `appSettings.json`.  It's not so easy in `web.config` of course, but perhaps there we just use some delimited string, that we parse to get the values, e.g.:

```
<add key="Umbraco.Forms.Integrations.EMerchantPay.Currencies" value="USD,UsDoller;EUR,Euro" />
```